proposals/0051-options-local-pragma.rst

```diff
@@ -0,0 +1,145 @@
+Options Local Pragma
+====================
+
+.. proposal-number:: 
+.. ticket-url::
+.. implemented::
+.. highlight:: haskell
+.. header:: This proposal is `discussed at this pull request <https://github.com/ghc-proposals/ghc-proposals/pull/0>`_.
+.. sectnum::
+.. contents::
+
+We propose to add functionality for switching GHC-options locally. For instance, this code
+::
+ {-# OPTIONS_GHC -Wno-name-shadowing #-}
+ 
+ f = {-# OPTIONS_LOCAL -Wname-shadowing #-}
+   let a = () in
+   let a = 5 in
+     a + a
+will result in a  ``-Wname-shadowing`` warning at line
+::
+ let a = 5 in
+
+This proposal rises some ticket problems:
+ 1. https://gitlab.haskell.org/ghc/ghc/issues/602
+ 2. https://gitlab.haskell.org/ghc/ghc/issues/10150
+
+Motivation
+------------
+
+There are the following reasons to implement this pragma.
+
+Local control of warnings
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To allow control of warnings when you are confident in your decisions without missing really important warnings.
+
+Warnings are compiler recomendations and often they can be really useful. They allow us to avoid stupid mistakes and point to *weak* places in the code. You should pay attention to them. But sometimes you can be confident in your local code-decision and get some useless warning. It clutters the warnings list and you can miss a really important warning.
```

Comment:
I would use neutral language in a proposal and avoid emotionally strong words like "stupid", telling the reader what to they "should" do, using the word "really", etc. I don't insist of course, it's your proposal, but be prepared that (some) people approach proposals with a skepticism.

For example, you say that warnings can be "really useful" but this is a bold statement that should be followed by an example of code where a warning catches a tricky bug or makes an insightful recommendation. (I would simply avoid saying things like this and get straight to the point)